src/transforms/addImportedSchemas.ts

```diff
@@ -0,0 +1,67 @@
+import * as path from "path";
+import * as fs from "fs";
+import * as ts from "typescript";
+import {
+  DefinitionNode,
+  DocumentNode,
+  GraphQLError,
+  Kind,
+  parse,
+} from "graphql";
+import { TypeContext } from "../TypeContext";
+import {
+  DiagnosticsWithoutLocationResult,
+  graphQlErrorToDiagnostic,
+} from "../utils/DiagnosticError";
+import { err, ok } from "../utils/Result";
+
+export function mergeImportedSchemas(
+  ctx: TypeContext,
+  doc: DocumentNode,
+): DiagnosticsWithoutLocationResult<DocumentNode> {
+  const importedSchemas: Set<string> = new Set();
+  for (const name of ctx.allNameDefinitions()) {
+    if (name.externalImportPath) {
+      importedSchemas.add(name.externalImportPath);
+    }
+  }
+  const importedDefinitions: DefinitionNode[] = [];
+  const errors: ts.Diagnostic[] = [];
+  for (const schemaPath of importedSchemas) {
+    const text = fs.readFileSync(path.resolve(schemaPath), "utf-8");
+    try {
+      const parsedAst = parse(text);
+      for (const def of parsedAst.definitions) {
+        if (
+          def.kind === Kind.OPERATION_DEFINITION ||
+          def.kind === Kind.FRAGMENT_DEFINITION ||
+          def.kind === Kind.DIRECTIVE_DEFINITION ||
```

Comment:
Why are we filtering this out? I think the schema might not be valid if its missing schema directives that are used.